rust/ql/lib/codeql/rust/elements/internal/PathImpl.qll

```diff
@@ -34,4 +34,24 @@ module Impl {
     pragma[nomagic]
     string getText() { result = this.getPart().getNameRef().getText() }
   }
+
+  /** A simple identifier path. */
+  class IdentPath extends Path {
+    private string name;
+
+    IdentPath() {
+      not this.hasQualifier() and
+      exists(PathSegment ps |
+        ps = this.getPart() and
+        not ps.hasGenericArgList() and
+        not ps.hasParenthesizedArgList() and
+        not ps.hasPathType() and
+        not ps.hasReturnTypeSyntax() and
+        name = ps.getNameRef().getText()
+      )
```

Comment:
Seems like it should be possible to share some code with the characteristic predicate of `VariableAccessCand` here. Perhaps by doing `_name = this.(PathExpr).getPath().(IdentPath).getName()` in there?

Reply:
This PR does exactly that :-)

Reply:
Lol, sorry 😂